src/promptflow/promptflow/_core/tracer.py

```diff
@@ -288,19 +296,53 @@ def enrich_span_with_input(span, input):
 
 
 def enrich_span_with_trace_type(span, inputs, output, trace_type):
-    enrich_span_with_output(span, output)
     if trace_type == TraceType.LLM:
         token_collector.collect_openai_tokens(span, output)
     elif trace_type == TraceType.EMBEDDING:
         token_collector.collect_openai_tokens(span, output)
         enrich_span_with_embedding(span, inputs, output)
     enrich_span_with_openai_tokens(span, trace_type)
+    return enrich_span_with_output(span, output)
+
+
+def traced_generator(generator, parent_span):
```

Comment:
The span is not actually `parent`. How about `linked_span`? or `original_span` or some other better name?